src/main/java/ilex/util/Logger.java

```diff
@@ -128,12 +138,49 @@ public static void setLogger( Logger logger )
 	* @return the singleton Logger
 	*/
 	public static Logger instance( )
+	{
+		return instance("");
+	}
+
+	/**
+	 * Get instance of logger. Create with given process name
+	 * if it doesn't exist yet. 
+	 * Process name will not be reset on subsequent calls.
+	 * @param procName
+	 * @return
+	 */
+	public static Logger instance(String procName)
 	{
 		if (theLogger == null)
-			theLogger = new StderrLogger("");
+		{			
+			theLogger = initializeLogger(procName);
+		}
 		return theLogger;
 	}
 
+	private static Logger initializeLogger(String procName)
+	{
+		try {
+			preventChange = System.getProperty("opendcs.logger",null) != null;
+			String loggerClassName = System.getProperty("opendcs.logger", "ilex.util.StderrLogger");			
+			System.err.println("Loading Logger Class " + loggerClassName);
+			Class<?> loggerClass = Class.forName(loggerClassName);
+			Constructor<?> constructor = loggerClass.getDeclaredConstructor(String.class);
+			Logger logger = (Logger)constructor.newInstance(procName);
+			logger.info(String.format("Logger %s started",loggerClassName));
+			return logger;
+		} catch (ClassNotFoundException e) {
+			throw new RuntimeException("Unable to initialize logging system. Logger not found.",e);
```

Comment:
Worth adding in the system property here: 
`"Unable to initialize logging system. Logger not found for class specified with opendcs.logger system property"`